addons/mrp/models/mrp_abstract_workorder.py

```diff
@@ -0,0 +1,366 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, fields, models, _
+from odoo.addons import decimal_precision as dp
+from odoo.exceptions import UserError
+from odoo.tools import float_compare, float_round, float_is_zero
+
+
+class MrpAbstractWorkorder(models.AbstractModel):
+    _name = "mrp.abstract.workorder"
+    _description = "basic features and fields shared between production wizard\
+    and workorders"
+
+    production_id = fields.Many2one('mrp.production', 'Manufacturing Order')
+    product_id = fields.Many2one(related='production_id.product_id', readonly=True, store=True)
+    qty_producing = fields.Float(string='Currently Produced Quantity', digits=dp.get_precision('Product Unit of Measure'))
+    product_uom_id = fields.Many2one('uom.uom', 'Unit of Measure', readonly=True)
```

Comment:
maybe put the onchange with the domain of the uom's category in the abstract model?